src/reader.rs

```diff
@@ -2511,6 +2511,16 @@ impl ReaderData {
                     self.update_buff_pos(elt, Some(el.position() + 1));
                 }
             }
+            rl::ForwardCharPassive => {
+                let (elt, el) = self.active_edit_line();
+                if self.is_navigating_pager_contents() {
+                    // Do nothing
```

Comment:
> > Even more ideally, fish would add a function that has semantics of
> > "move cursor to the cell number X on line number Y, where Y is counted
> > from the line that starts the prompt".
> 
> The first seems eminently doable. The second might be a more difficult ask as things currently stand.

So the only difficulty is that move(X, Y) needs to potentially skip over Y-1 soft-wrapped lines?
We can probably use the current rendering if we want to do that.
If the current prompt rendering exceeds `$LINES`, does it still start from the prompt, or the first visible character?
Probably the first. Either way this scenario has always been somewhat broken in fish.

Anyway, making it work with something like `forward-char` seems more conventional here [*].

I'd expect that we'll eventually want a forward-char-like command that does not move in the pager and probably also not accept autosuggestion.
Crucially, when the pager is active, it should move the cursor inside the pager search field (which is impossible today).
This was suggested in https://github.com/fish-shell/fish-shell/issues/10268#issuecomment-1913142361;
With the recent input event queueing changes this might be possible now.
Quick sketch:

```
bind \cf '
	if commandline --paging-mode
	    commandline -f move-right
	else if commandline --is-at-end
	    commandline -f accept-autosuggestion-char
	end
	    commandline -f forward-char # different name to avoid breakage?
	end
'
```

I'm not sure if the autosuggestion part makes sense.
In general, it's not clear where the line should be between scripts and builtin behavior.

[*]: Note that currently we use this to move to a X/Y coordinates (logical, ignoring soft-wrapping).  Maybe it's worth making this more ergonomic.

```
    commandline -C 0
    for _line in (seq $line)[2..]
        commandline -f down-line
    end
    commandline -f beginning-of-line
    for _column in (seq $column)[2..]
        commandline -f forward-single-char
    end
```

Reply:
If we take `forward-char-passive`, I expect it should also move the cursor in paging mode (in the pager search field). I'm not sure what kitty integration expects.
Given that the `forward-char` naming is nicer for rebinding that the above if/else command,
it seems fine to have an overlapping low-level command. Perhaps we can find some kind of naming scheme for low-level commands.